tst/EnergyPlus/unit/ScheduleManager.unit.cc

```diff
@@ -804,3 +806,445 @@ TEST_F(EnergyPlusFixture, Schedule_GetCurrentScheduleValue_DST)
+        "  For: AllDays,            !- Field 2",
+        "  Until: 24:00,            !- Field 3",
+        "  1,                       !- Field 4",
+        "  Through: 12/31,          !- Field 5",
+        "  For: AllDays,            !- Field 6",
+        "  Until: 24:00,            !- Field 7",
+        "  2;                       !- Field 8",
+    });
+
+    ASSERT_TRUE(process_idf(idf_objects));
+
+    DataGlobals::NumOfTimeStepInHour = 4;    // must initialize this to get schedules initialized
+    DataGlobals::MinutesPerTimeStep = 15;    // must initialize this to get schedules initialized
+    DataGlobals::TimeStepZone = 0.25;
+    DataGlobals::TimeStepZoneSec = DataGlobals::TimeStepZone * DataGlobals::SecInHour;
+
+    ScheduleManager::ProcessScheduleInput(state.outputFiles); // read schedules
+
+    DataEnvironment::RunPeriodStartJDay = 1;
+
+    DataEnvironment::Month = 12;
+    DataEnvironment::DayOfMonth = 31;
+    DataGlobals::HourOfDay = 24;
+    DataEnvironment::DayOfWeek = 4;
+    DataEnvironment::DayOfWeekTomorrow = 5;
+    DataEnvironment::HolidayIndex = 0;
+    DataGlobals::TimeStep = 1;
+    DataEnvironment::DayOfYear_Schedule = General::OrdinalDay(DataEnvironment::Month, DataEnvironment::DayOfMonth, 1);
+
+    DataEnvironment::DSTIndicator = 0; // DST IS OFF
+    ScheduleManager::UpdateScheduleValues();
+    EXPECT_EQ(2.0, ScheduleManager::LookUpScheduleValue(1, DataGlobals::HourOfDay, DataGlobals::TimeStep));
+    EXPECT_EQ(2.0, ScheduleManager::Schedule(1).CurrentValue);
+    EXPECT_EQ(2.0, ScheduleManager::GetCurrentScheduleValue(1));
+
+    DataEnvironment::DSTIndicator = 1; // DST IS ON
+    ScheduleManager::UpdateScheduleValues();
+    // Since DST is on, you're actually on the next day, which in this specific case should be 1/1 at 0:15
+    // so it **should** return 1.0
+    EXPECT_EQ(1.0, ScheduleManager::Schedule(1).CurrentValue);
+    EXPECT_EQ(1.0, ScheduleManager::GetCurrentScheduleValue(1));
+    EXPECT_EQ(1.0, ScheduleManager::LookUpScheduleValue(1, DataGlobals::HourOfDay, DataGlobals::TimeStep));
+}
+
+
+TEST_F(EnergyPlusFixture, Schedule_GetCurrentScheduleValue_DST_RampUp_Leap) {
+
+    // So here we'll mimic using a Schedule:Compact that ramps up constantly
+
+    // Schedule:Compact,
+    //   RampingUp,                 !- Name
+    //   Any Number,                !- Schedule Type Limits Name
+    //   Through: 1/1,
+    //   For: AllDays,
+    //    Until: 1:00, 1,
+    //    Until: 2:00, 2,
+    //    [...]
+    //    Until: 24:00, 24,
+    //   Through: 1/2,
+    //   For: AllDays,
+    //    Until: 1:00, 25,
+    //    Until: 2:00, 26,
+    //    Until: 3:00, 27,
+    //    [...]
+    //   Through: 12/31,
+    //   For: AllDays,
+    //    Until: 1:00, 8761,
+    //    [...]
+    //    Until: 24:00, 8784;
+
+    // # 'THROUGH" => Number of additional week schedules
+    // # 'FOR' => Number of additional day schedules
+    // So we use 366 Week Schedules, all with one day (LeapYear)
```

Comment:
Test a ramp up schedule for 8784 hourly values (leap year), checking every hour in the DST and non DST case.